packages/fast-components-react-base/src/stack-panel/stack-panel.props.ts

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import {
+    ManagedClasses,
+    StackPanelClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-base";
+
+export interface StackPanelManagedClasses
+    extends ManagedClasses<StackPanelClassNameContract> {}
+export interface StackPanelUnhandledProps extends React.HTMLAttributes<HTMLDivElement> {}
+
+export enum stackPanelOrientation {
+    horizontal = "horizontal",
+    vertical = "vertical",
+}
+
+export interface ItemSpanOverride {
+    [key: number]: number;
+}
+
+export interface StackPanelHandledProps extends StackPanelManagedClasses {
+    /**
+     * Whether items outside the viewport should be rendered or not
+     * default is true
+     */
+    enableVirtualization?: boolean;
+
+    /**
+     *  the indexes of children that should always be rendered
+     */
+    neverVirtualizeIndexes?: number[];
+
+    /**
+     *  The default span of each item (width in horizontal orientation, height in vertical)
+     */
+    defaultItemSpan?: number;
+
+    /**
+     * An object whose entries specifies the height of items that differ from the default height.
+     * Format is {itemIndex: heightValue}
+     */
+    itemSpanOverrides?: ItemSpanOverride;
+
+    /**
+     * How many items to preload before and after the range defined by viewportMaxHeight
+     */
+    preloadBufferLength?: number;
+
+    /**
+     * The orientation of the stackpanel
+     */
+    orientation?: stackPanelOrientation;
+
+    /**
+     * The children of the panel
+     */
+    children?: React.ReactNode;
+
+    /**
+     * Callback for when scroll values change
+     */
+    onScrollChange?: (newScrollValue: number, scrollMaxValue: number) => void;
+
+    /**
+     * When the component scrolls focused content into view how many pixels should the next item "peek" into view
+     */
+    nextItemPeek?: number;
```

Comment:
should we instead of offering a peek, just make this controlled with regards to the scrolled distance?

Reply:
removed animation code as I found a css based solution